tendrl/common/manager/job_sync.py

```diff
@@ -0,0 +1,145 @@
+from tendrl.common.flows.exceptions import FlowExecutionFailedError
+
+LOG = logging.getLogger(__name__)
+
+
+class EtcdRPC(object):
+
+    def __init__(self, syncJobThread):
+        self.config = syncJobThread._manager._config
+        etcd_kwargs = {
+            'port': int(self.config.get("common", "etcd_port")),
+            'host': self.config.get("common", "etcd_connection")
+        }
+
+        self.client = etcd.Client(**etcd_kwargs)
+        self.syncJobThread = syncJobThread
+
+    def _process_job(self, raw_job, job_key, job_type):
+        # Pick up the "new" job that is not locked by any other integration
+        if raw_job['status'] == "new" and raw_job["type"] == job_type:
+            raw_job['status'] = "processing"
+            # Generate a request ID for tracking this job
+            # further by tendrl-api
+            req_id = str(uuid.uuid4())
+            raw_job['request_id'] = "%s/flow_%s" % (
+                self.syncJobThread._manager.integration_id, req_id)
+            self.client.write(job_key, json.dumps(raw_job))
+            try:
+                definitions = self.validate_flow(raw_job)
+                if definitions:
+                    self.invoke_flow(raw_job['run'], raw_job, definitions)
+            except FlowExecutionFailedError as e:
+                LOG.error(e)
+                raw_job['status'] = "failed"
+            else:
+                raw_job['status'] = "finished"
+
+            return raw_job, True
+        else:
+            return raw_job, False
+
+    def _acceptor(self, job_type):
+        while not self.syncJobThread._complete.is_set():
+            jobs = self.client.read("/queue")
+            for job in jobs.children:
+                executed = False
+                if job.value is None:
+                    continue
+                raw_job = json.loads(job.value.decode('utf-8'))
+                try:
+                    raw_job, executed = self._process_job(
+                        raw_job,
+                        job.key,
+                        job_type
+                    )
+                    if "etcd_client" in raw_job['parameters']:
+                        del raw_job['parameters']['etcd_client']
+                except FlowExecutionFailedError as e:
+                    LOG.error(e)
+                if executed:
+                    self.client.write(job.key, json.dumps(raw_job))
+                    break
+            gevent.sleep(2)
+
+    def run(self):
+        self._acceptor(self.syncJobThread._manager.name)
+
+    def stop(self):
+        pass
+
+    def validate_flow(self, raw_job):
+        definitions = yaml.load(
+            self.client.read(
+                self.syncJobThread._manager.defs_dir
+            ).value.decode("utf-8")
+        )
+        definitions = DefinitionsSchemaValidator(
+            definitions).sanitize_definitions()
+        # resp, msg = JobValidator(definitions).validateApi(raw_job)
```

Comment:
Will be enabled once validator is corrected and works fine. @TimothyAsir is working on the same.